### DISCUSSION TITLE
Add custom icons / community page to add icons for other 2FA applications

### DISCUSSION THREAD

**Discussion by Danek309044** [ente-io/ente#3409] [Enhancements]
So for example, as a user of Uptime Robot I see in the Ente Auth just "U" as the app icon and that is like for me a little harder to spot beacuse the Uptime Robot icon is a green circle. So there are 2 ways that I think we could add custom icons:
1. We would uplaod our custom icons to Ente cloud and then we use them in Ente Auth
2. Some community website thing where you would upload it and other users could use it

**Comment by desaianand1** [comment]
Yes, 100%

So many services could benefit from this or even just scraping the websites favicon

**Reply by vishnukvmd** [reply]
Scraping the website for favicon unfortunately has privacy implications, which is why we've not yet implemented this.

Making API calls directly to them complicates our privacy policy, adding a proxy in between means we get to know which services you are using 2FA for.

**Reply by EweSparky** [reply]
But conversely if I make a pull request with an icon for _example.com_, then everyone knows I use it. No easy solution!

I feel like either a no-log proxy where you download it for us and we download from you, or us being able to specify a domain for the app to scrape from is the better compromise. At the end of the day, if I already use _example.com_, me scraping the logo isn't going to negatively impact my privacy any more than it already has been. 

Alternatively, it'd be great if I could upload my own logo direct to the app and have it backup that data with the rest of my E2EE configuration.

**Comment by iedame** [comment]
It is easy to contribute custom icons through GitHub. Do you mean a more user-friendly way to contribute for those that don't know how git works?

**Reply by Cybertronus** [reply]
What is the process for submitting new icons? Is it documented somewhere?
There used to be a [doc](https://github.com/ente-io/ente/blob/main/auth/docs/adding-icons.md) (indexed by Google), but no longer available!

**Reply by iedame** [reply]
The correct path for that doc file is this one now:
https://github.com/ente-io/ente/blob/main/mobile/apps/auth/docs/adding-icons.md
If you still have any questions let us know.

**Reply by Danek309044** [reply]
Ahh i didnt even saw that. Thank you for letting me know
